### PR TITLE
DS-123 add env vars for API server

### PR DESCRIPTION
**Pull request by milt**
[DS-123] API server should have a configurable root prefix, host, port etc.

[DS-123]: https://yet.atlassian.net/browse/DS-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by kelvinqian00** [comment]
No plans for a config file?

File: src/server/com/yetanalytics/datasim/server.clj [hunk 1]
```diff
@@ -240,29 +241,48 @@
...
+  (let [root-path       (get env :api-root-path "")
+        _               (when (not-empty root-path)
+                          (when-not (str/starts-with? root-path "/")
+                            (throw (ex-info "API_ROOT_PATH must start with /"
+                                            {:type      ::invalid-root-path
+                                             :root-path root-path})))
+                          (when (str/ends-with? root-path "/")
+                            (throw (ex-info "API_ROOT_PATH must not end with /"
+                                            {:type      ::invalid-root-path
+                                             :root-path root-path}))))
```
Comment (kelvinqian00) [inline src/server/com/yetanalytics/datasim/server.clj:255]: Probably worth separating this out into its own assert function.

Comment (milt) [reply src/server/com/yetanalytics/datasim/server.clj]: https://github.com/yetanalytics/datasim/pull/72/commits/7ffb0016330611f9374fc18ab48e881db9cda93a

File: README.md [hunk 2]
```diff
@@ -188,19 +188,17 @@ By default the server starts at http://localhost:9090
...
+| CREDENTIALS         | <none>                                                 | Basic Authentication credentials required to call the API endpoints in the form of `username:password` | `datasim:datasim` |
+| API_ROOT_PATH       | <none>                                                 | Root path to prefix API routes. Must begin with a `/`, cannot end with a `/`.                          | `/foo`            |
+| API_HOST            | `0.0.0.0`                                              | Host on which to bind the API server.                                                                  | `localhost`       |
+| API_PORT            | `9090`                                                 | Port on which to bind the API server.                                                                  | `8080`            |
+| API_ALLOWED_ORIGINS | `https://yetanalytics.github.io,http://localhost:9091` | CORS allowed origins for the API server, separated by commas.                                          | `*`               |
```
Comment (kelvinqian00) [inline README.md:199]: You should use the Markdown trick Emma showed us last year to hide this URL:
```
<details>https://yetanalytics.github.io,http://localhost:9091<summary>(URLs)</summary></details>
```

**Review by kelvinqian00 — APPROVED** [review]

**Pull request merged**